### PR TITLE
Properly filter issue list given no assignees filter

### LINKED ISSUE #31520: Issues belonging to a repository cannot be filtered by "no assignee"
### Description

While attempting to filter issues within a repository, issues cannot be filtered by the "no assignee" criteria. This makes it difficult to locate issues that are not assigned to any specific individual.

### Gitea Version

1.22.0

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

MacOS - Chrome

### How are you running Gitea?

Using official docker image

### Database

None

### PR DESCRIPTION
**Pull request by kemzeb**
Quick fix #31520. This issue is related to #31337.

### CODE REVIEW

**Comment by kemzeb** [comment]
Example without the filter:

Screenshot 2024-06-28 at 9 59 34 AM

Example with:

Screenshot 2024-06-28 at 10 00 20 AM

File: modules/indexer/issues/dboptions.go [hunk 1]
```diff
@@ -44,6 +44,12 @@ func ToSearchOptions(keyword string, opts *issues_model.IssuesOptions) *SearchOp
 		searchOpt.ProjectID = optional.Some[int64](0) // Those issues with no project(projectid==0)
 	}
 
+	if opts.AssigneeID > 0 {
+		searchOpt.AssigneeID = optional.Some(opts.AssigneeID)
+	} else if opts.AssigneeID == -1 { // FIXME: this is inconsistent from other places
+		searchOpt.AssigneeID = optional.Some[int64](0)
```
Comment (kemzeb) [inline modules/indexer/issues/dboptions.go:50]: Would there ever be an user with an ID of 0?

**Comment by kemzeb** [comment]
I also decided to test the other issue filtering options to see if similar problems occur and they all appear to work fine.

**Review by silverwind — APPROVED** [review]

**Comment by lunny** [comment]
Could you also have a test for it to avoid the bug being occupied again?

**Comment by kemzeb** [comment]
Yep, I'll see if I can write a regression test for this today

**Comment by kemzeb** [comment]
Not sure if its the best place, but I added the regression test to `issues/indexer_test.go` and ran it on my branch and current main

**Review by lunny — APPROVED** [review]

**Pull request merged**